src/reader.rs

```diff
@@ -276,7 +279,10 @@ impl<R: Read + Seek> VdbReader<R> {
                         "Skipping blosc decompression because of a {}-count read",
                         count
                     );
-                    vec![T::zeroed(); 0]
+                    {
+                        T::zeroed();
+                        vec![] as std::vec::Vec<T>
+                    }
```

Comment:
It looks like your clippy message for this is:

```rust
warning: function or method calls as the initial value in zero-sized array initializers may cause side effects
   --> src/reader.rs:282:21
    |
282 |                     vec![T::zeroed(); 0]
    |                     ^^^^^^^^^^^^^^^^^^^^ help: consider using: `{ T::zeroed(); vec![] as std::vec::Vec<T> }`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#zero_repeat_side_effects
    = note: `#[warn(clippy::zero_repeat_side_effects)]` on by default
```

Which, when translated, simply says that Rust _might never execute `T::zeroed()` at all_ if it sees that the resulting `Vec` is going to be zero. Since `T::zeroed()` is not performing any _side effects_, we can should **discard this odd suggestion** and just use the empty vector constructor again (if it requires a type annotation, `Vec::<T>::new()` is your friend).

```suggestion
                    vec![]
```